packages/core/src/RenderingEngine/VideoViewport.ts

```diff
@@ -265,8 +387,101 @@ class VideoViewport extends Viewport implements IVideoViewport {
     });
   }
 
+  public getScalarData() {
+    const canvas = document.createElement('canvas');
+    canvas.width = this.videoWidth;
+    canvas.height = this.videoHeight;
+    const context = canvas.getContext('2d');
+    context.drawImage(this.videoElement, 0, 0);
+    const canvasData = context.getImageData(
+      0,
+      0,
+      this.videoWidth,
+      this.videoHeight
+    );
+    const { data: scalarData } = canvasData;
+    (scalarData as any).getRange = () => [0, 255];
+    return scalarData;
+  }
```

Comment:
we don't have getScalarData for the rest of the viewports, is this really necessary to have it as public? getImageData is enough imo